### DISCUSSION TITLE
Custom status page for custom scripts - Looking for possibilities

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#13559] [General]
Hi

I am looking for something that will allow me to see the status of each of the scripts we run each day. A nice little overview, if you will, inside Netbox.
It's nothing fancy, just something that will tell me if script X-test has run successfully today. 

To elaborate:
We currently use a custom field for Netbox devices (DCIM) which shows if backup has run ok or not and another field with a timestamp, for when the backup was last run.
That was the first script we employed. Now we have multiple scripts running, collecting all sorts of data from our production environment and putting that data into Netbox, in one way or the other, so custom field associated with devices/vm's would not be good.
I am looking for a way to easily see if these scripts have run and what the result of the run was (OK/NotOK). I would prefer this to be inside Netbox somehow, because we look there all the time anyway, but I am open to other suggestions.
These scripts are not run within Netbox, but python-scripts on another server, populating Netbox with a multitude of various data.

Would a simple webpage on a server be better, than incorporating something like this into Netbox?

**Comment by candlerb** [comment]
> I am looking for a way to easily see if these scripts have run and what the result of the run was (OK/NotOK)

From my point of view, this is an issue of monitoring, not configuration - and Netbox is primarily intended as a configuration database, not as a portal onto monitoring systems.  Therefore, I'd recommend you monitor these scripts using whatever your current or preferred monitoring system is.

For example: personally I really like Prometheus.  You could set up your batch jobs to send their success/failure result and their execution time as metrics to the prometheus "pushgateway"; this then exposes these values to be scraped by Prometheus at regular intervals, and then you could expose these in a dashboard, e.g. using Grafana.

Note that prometheus isn't an events system, only a metrics system. You can't use it to store *all* the script result runs, but you can expose the most recent one to be scraped (which means if you run multiple executions within one scrape interval, only the most recent one will be stored). You can also store metrics which summarise all the runs, e.g. counters for successful and failed executions.

If you want to store and query the status of individual execution runs, then a logging system like Loki or VictoriaLogs may suit you better.
